src/engine/serverbrowser.cpp

```diff
@@ -346,63 +323,34 @@ void checkpings()
     ENetBuffer buf;
     ENetAddress addr;
     uchar ping[MAXTRANS];
-    char text[MAXTRANS];
     buf.data = ping;
     buf.dataLength = sizeof(ping);
     while(enet_socket_wait(pingsock, &events, 0) >= 0 && events)
     {
         int len = enet_socket_receive(pingsock, &addr, &buf, 1);
         if(len <= 0) return;
         serverinfo *si = NULL;
-        loopv(servers) if(addr.host == servers[i]->address.host && addr.port == servers[i]->address.port) { si = servers[i]; break; }
-        if(!si && searchlan) si = newserver(NULL, addr.port-1, 1, NULL, NULL, NULL, NULL, addr.host);
-        if(!si) continue;
-        ucharbuf p(ping, len);
-        int millis = getint(p), rtt = clamp(totalmillis - millis, 0, min(serverdecay*1000, totalmillis));
-        if(millis >= lastreset && rtt < serverdecay*1000) si->addping(rtt, millis);
-        si->lastinfo = totalmillis;
-        si->numplayers = getint(p);
-        int numattr = getint(p);
-        si->attr.shrink(0);
-        loopj(numattr) si->attr.add(getint(p));
-        int gver = si->attr.empty() ? 0 : si->attr[0];
-        getstring(text, p);
-        filterstring(si->map, text, false);
-        getstring(text, p);
-        filterstring(si->sdesc, text, true, true, true, false, MAXSDESCLEN+1);
-        si->players.deletearrays();
-        si->handles.deletearrays();
-        if(gver >= 227)
-        {
-            getstring(text, p);
-            filterstring(si->branch, text, true, true, true, false, MAXBRANCHLEN+1);
-        }
-        loopi(si->numplayers)
-        {
-            if(p.overread()) break;
-            getstring(text, p);
-            si->players.add(newstring(text));
-        }
-        if(gver >= 225) loopi(si->numplayers)
+        loopv(servers)
         {
-            if(p.overread()) break;
-            getstring(text, p);
-            si->handles.add(newstring(text));
+            if( servers[i]->is_same( addr ) )
+            {
+                si = servers[i];
+                break;
+            }
         }
-        sortedservers = false;
+        if(!si && searchlan) si = serverinfo::newserver(NULL, addr.port-1, 1, NULL, NULL, NULL, NULL, addr.host);
```

Comment:
TODO for self: check what `newserver` does (likely some sort of factory).

Reply:
Please use explicit pointer checks, especially in new code. That helps increase the readability.